### DISCUSSION TITLE
osu!mania Argon skin middle column size should match others

### DISCUSSION THREAD

**Discussion by kadoen** [ppy/osu#29733] [Ideas]
This is one of the points mentioned in #24619, but I think it's a relatively simple change that could be implemented quickly.
Currently, the middle column in Argon skin (usually spacebar) is much wider than the rest, which makes it uncomfortable to read, especially in streams and similar patterns. For comparison, see image below, the same pattern with a even-column skin and current argon.
Even if it is the spacebar, it is generally not treated as a special column in mapping, and **it should be the same width as the other columns**. It being a different colour than the rest is good, and should be kept.

![comparison](https://github.com/user-attachments/assets/6c287d77-3e28-4db1-9fb2-03f7f6832622)

Related to the above, **special style (N+1) keys could be made wider than the rest**, similar to the current middle column, emulating beatmania/ez2dj scratch keys ([example video](https://www.youtube.com/watch?v=NJpN2sBB_TY&t=35)). That is something a lot of skins do, because there that column really has a different meaning in mapping. Ideally it should follow the "Use special (N+1) style" toggle in map details, but I don't know if that is feasible to make relate to a skinning option. In that case, the 7k part should follow the same color scheme as standalone 7k, with the special column being a different colour. 

In any case, this second point could be another discussion, my main goal is raising the issue in middle column widths.

**Comment by Tenexxt** [comment]
I personally like the middle column being bigger, but cosmetically it would be nice to make it possible to make certain keys wider/thiner.
Since it's being worked on so you can move around and rotate the mania playfield, making keys wider also would make sense to be added _In My Opinion_.
This would allow for faster skin editing for custom skins, and would be definitely useful for people who just on argon.

This part might be kinda off topic, but also it should be possible to make the columns different colours for people that just like the stable ones and give column-specific settings based on key count
(Like making it possible for the skin to save that on 5k the middle column should be wide, but on 7k it's thin)

This is a very ambitious suggestion which because of its complexity won't be added, but I personally think that would be useful.

**Reply by kadoen** [reply]
It's already possible to make certain keys wider, make it column-specific, and change the colours/sprites. But I don't think the default skin should have that column wide as it is, no 7k player plays like that.

**Reply by Tenexxt** [reply]
I meant for the argon skin render, and since argon is rendered by the game, not a sprite, it's impossible to export the sprites from the game directly like with other custom skins.
#29530 is the same thing (for colours) that I'm talking about, but that would be also editable in game, not just through skin file.

As for making the middle key smaller, I'd personally prefer for it to stay, but making a suggestion to make it into a trigger would trigger the devs from making all the unnecessary stuff when it can be decided - imo a vote should be made **just in case** if it should be changed or not.

**Reply by cl8n** [reply]
> Since it's being worked on so you can move around and rotate the mania playfield, making keys wider also would make sense to be added In My Opinion.

for sure, and I'd assume that's already in mind as a long-term goal, but this issue is just suggesting better defaults.

**Reply by Tenexxt** [reply]
As I already said, a poll could be made on whatever the key should stay wider or it should be chopped, so it would ensure the majority is going to be happy (that of course would be irrelevant if in some time [years] we would get the option to edit the properties of individual keys without making a seperate skin)

**Reply by kadoen** [reply]
A poll can be made if that's what it's preferred from the dev team, but no player or skin I've ever seen has this feature. That's why it was mentioned to smoogi last year by a group of community members, and I'm bringing it up again here. 
Especially since this skin is what's used in official videos, which have a much larger reach than just lazer players.

**Reply by peppy** [reply]
Wait for this to be adjustable in the skin editor.

**Reply by kadoen** [reply]
Reading recent threads, it seems like argon skinning is coming soon-ish (may be wrong), so I wanted to bump this thread to consider this when setting the default parameters for the argon skin in mania, as I still believe the wide central column to be an issue.

**Comment by Netzoid** [comment]
I kinda agree. A middle column being bigger would emphasize that specific column more but in most maps, every column in about the same in note count and emphasis so it would be quite odd to make one column bigger; it just gives the player the sense that this column is “better” than the others. Furthermore, from my playing perspective it can be a bit hard to get used to especially if you're transitioning from stable where majority of skins make every column the same. Since N+1 keys is the most emphasized columns, they should definitely have bugger columns than the rest, not keys without any sort of emphasis built into them. 

However, I do believe it’s important to differentiate the middle columns from the other ones. I just think it’s way too wide. I think it would be better if it was just slightly wider than the normal notes

**Reply by kadoen** [reply]
Most skins do have that center column in a different colour than the rest to make it stand out, see my left image with the o2jam-style colouring. That is usually enough to make it differentiable, but it could be changed to a more flashy colour as well.
